v2/devices/beryllium.yml

```diff
@@ -0,0 +1,225 @@
+name: "Xiaomi Poco F1"
+codename: "beryllium"
+formfactor: "phone"
+aliases: []
+doppelgangers: []
+user_actions:
+  bootloader:
+    title: "Reboot to Bootloader"
+    description: "With the device powered off, hold Volume Down + Power."
+    image: "phone_power_down"
+    button: true
+  recovery:
+    title: "Reboot to Recovery"
+    description: "With the device powered off, hold Volume Up + Power."
+    image: "phone_power_up"
+    button: true
+  boot:
+    title: "Boot the device"
+    description: "Power on the device."
+    image: "phone_power_up"
+    button: true
+  confirm_model:
+    title: "Confirm your model"
+    description: "Please double-check that your device is a Xiaomi Poco F1 (beryllium)."
+  unlock:
+    title: "OEM unlock"
+    description: "If you haven't done so already, make sure to OEM unlock your device first."
+    link: "https://en.miui.com/unlock/"
+  warning:
+    title: "Warning"
+    description: "The port is based on Android 9 firmware (V11.0.5.0.PEJMIXM) and Lineage 16 vendor image. Current Recovery will be replaced with Ubports Recovery! Remember to flash the latest firmware when reverting to latest Android builds!"
```

Comment:
> Also, the disadvantage of forcing a certain firmware is that people that might have upgraded radio or other firmware to a newer version could accidentially revert their firmware. So I am not a big fan of automatic firmware flashing by us. I think its better to force the user to do that by themselves. Also, for copyright reasons, hosting firmware poses a certain risk that it will be taken down at some point.

I understand your point. I did add a warning part for firmware downgrade in the installer like this. As for firmware URL take down, they are generally packaged by `https://xiaomifirmwareupdater.com/projects/xiaomi-flashable-firmware-creator/` and the website hosts them. Also, most Poco F1 android 10+ zip present with error if firmware is too old. Since these firmware files are already redistributed by other websites, I think its fine being accessible from some other source too. If anything has a chance of getting take down, its that website mostly. So having a copy in repo is fine I thought. Also, if users have to manually download files, copy them to device, use custom recovery to flash, there would be less reason to have installer as the only three other steps to do are flash the system, boot and vendor. So i think i ll have firmware link up in my repo and have the user do manual download from the installer. Is that fine?